src/common/cluster-ipc.ts

```diff
@@ -56,17 +59,42 @@ if (ipcMain) {
   });
 
   handleRequest(clusterDisconnectHandler, (event, clusterId: ClusterId) => {
-    appEventBus.emit({name: "cluster", action: "stop"});
+    appEventBus.emit({ name: "cluster", action: "stop" });
     const cluster = ClusterStore.getInstance().getById(clusterId);
 
-    if (cluster) {
-      cluster.disconnect();
-      clusterFrameMap.delete(cluster.id);
+    if (!cluster) {
+      return;
+    }
+
+    cluster.disconnect();
+    clusterFrameMap.delete(cluster.id);
+  });
+
+  handleRequest(clusterDeleteHandler, async (event, clusterId: ClusterId) => {
+    appEventBus.emit({ name: "cluster", action: "remove" });
+    const cluster = ClusterStore.getInstance().getById(clusterId);
+
+    if (!cluster) {
+      return;
+    }
+
+    cluster.disconnect();
+    clusterFrameMap.delete(cluster.id);
+    const kubectl = await cluster.ensureKubectl();
```

Comment:
We should use bundled kubectl for these operations.

Reply:
Always? okay.